pool/web/handler/admin_custom_fields.ml

```diff
@@ -0,0 +1,159 @@
+module HttpUtils = Http_utils
+module Message = HttpUtils.Message
+
+let create_layout req = General.create_tenant_layout `Admin req
+
+let boolean_fields =
+  Custom_field.boolean_fields |> CCList.map Pool_common.Message.Field.show
+;;
+
+let find_assocs_in_urlencoded urlencoded field encoder =
+  let field = Pool_common.Message.Field.show field in
+  CCList.filter_map
+    (fun (key, values) ->
+      let group, id = CCString.take_drop (CCString.length field) key in
+      match CCString.equal field group with
+      | false -> None
+      | true ->
+        let key =
+          let open CCOption in
+          id
+          |> CCString.chop_prefix ~pre:"["
+          >>= CCString.chop_suffix ~suf:"]"
+          >>= fun l -> l |> encoder
+        in
+        let value = CCList.head_opt values in
+        (match key, value with
+         | Some l, Some v -> Some (l, v)
+         | _ -> None))
+    urlencoded
+;;
+
+let id req field encode =
+  Sihl.Web.Router.param req @@ Pool_common.Message.Field.show field |> encode
+;;
+
+let index req =
+  let open Utils.Lwt_result.Infix in
+  let result ({ Pool_context.tenant_db; _ } as context) =
+    Lwt_result.map_error (fun err -> err, "/admin/dashboard")
+    @@ let%lwt field_list = Custom_field.find_all tenant_db () in
+       Page.Admin.CustomFields.index field_list context
+       |> create_layout ~active_navigation:"/admin/custom-fields" req context
+       >|= Sihl.Web.Response.of_html
+  in
+  result |> HttpUtils.extract_happy_path req
+;;
+
+let form ?id req =
+  let open Utils.Lwt_result.Infix in
+  let open Lwt_result.Syntax in
+  let result ({ Pool_context.tenant_db; _ } as context) =
+    Lwt_result.map_error (fun err -> err, "/admin/custom-fields")
+    @@
+    let flash_fetcher key = Sihl.Web.Flash.find key req in
+    let* custom_field =
+      match id with
```

Comment:
This seems to be something like `>|> CCOption.map_or ~default:(Lwt.return_none) Custom_field.find`